tests/test_date.py

```diff
@@ -34,6 +54,215 @@ def test_time_formats(self):
+
+    def test_minus(self):
+        date_1 = Date('jan 10')
+        date_1.microsecond = 1
+        date_2 = Date('jan 5')
+        date_2.microsecond = 1
+        self.assertEqual(Date(date_1) - '5 days', date_2)
+
+    def test_adjustment(self):
+        d = Date('Jan 1st 2014 at 10 am')
+        self.assertEqual(d.year, 2014)
+        self.assertEqual(d.month, 1)
+        self.assertEqual(d.day, 1)
+        self.assertEqual(d.hour, 10)
+        self.assertEqual(d.minute, 0)
+        self.assertEqual(d.second, 0)
+
+        d.hour = 5
+        d.day = 15
+        d.month = 4
+        d.year = 2013
+        d.minute = 40
+        d.second = 14
+        d.microsecond = 10001
+
+        self.assertEqual(d.year, 2013)
+        self.assertEqual(d.month, 4)
+        self.assertEqual(d.day, 15)
+        self.assertEqual(d.hour, 5)
+        self.assertEqual(d.minute, 40)
+        self.assertEqual(d.second, 14)
+        self.assertEqual(str(d.date), '2013-04-15 05:40:14.010001')
+
+    def test_next_prev(self):
+        # Past month
+        self.assert_date('nov', datetime(2017, 11, 1))
+        self.assert_date('this nov', datetime(2017, 11, 1))
+        self.assert_date('last nov', datetime(2016, 11, 1))
+        self.assert_date('previous nov', datetime(2016, 11, 1))
+        self.assert_date('next nov', datetime(2017, 11, 1))
+        self.assert_date('upcoming nov', datetime(2017, 11, 1))
+
+        # Future month
+        self.assert_date('feb', datetime(2018, 2, 1))
+        self.assert_date('this feb', datetime(2018, 2, 1), )
+        self.assert_date('last feb', datetime(2017, 2, 1))
+        self.assert_date('previous feb', datetime(2017, 2, 1))
+        self.assert_date('next feb', datetime(2018, 2, 1))
+        self.assert_date('upcoming feb', datetime(2018, 2, 1))
+
+        # Current month
+        self.assert_date('june', datetime(2017, 6, 1))
+        self.assert_date('this june', datetime(2017, 6, 1))
+        self.assert_date('last june', datetime(2016, 6, 1))
+        self.assert_date('next june', datetime(2018, 6, 1))
+
+        # Past weekday
+        self.assert_date('sun', datetime(2017, 6, 18))
+        self.assert_date('this sun', datetime(2017, 6, 18))
+        self.assert_date('last sun', datetime(2017, 6, 11))
+        self.assert_date('previous sun', datetime(2017, 6, 11))
+        self.assert_date('next sun', datetime(2017, 6, 18))
+        self.assert_date('upcoming sun', datetime(2017, 6, 18))
+
+        # Future weekday
+        self.assert_date('wed', datetime(2017, 6, 21))
+        self.assert_date('last wed', datetime(2017, 6, 14))
+        self.assert_date('previous wed', datetime(2017, 6, 14))
+        self.assert_date('next wed', datetime(2017, 6, 21))
+        self.assert_date('upcoming wed', datetime(2017, 6, 21))
+
+        # Current weekday
+        self.assert_date('fri', datetime(2017, 6, 16))
+        self.assert_date('this fri', datetime(2017, 6, 16))
```

Comment:
Should be next friday datetime(2017, 6, 23)

Reply:
As above.